source/mainnet/tutorials/using-ID-in-dApps/wallet-connectors-tutorial.rst

```diff
@@ -0,0 +1,672 @@
+
+Finally, let's handle potential errors:
+
+.. code-block:: javascript
+
+  catch (e) {
+    if (isWalletConnectError(e)) {
+      throw new Error(
+        "Generating proof request rejected in wallet: " + JSON.stringify(e),
+      );
+    }
+    throw e;
+  }
+
+The error handling section checks for WalletConnect-specific errors using the ``isWalletConnectError`` helper function. This allows us to provide more specific error messages when the proof request is rejected by the wallet.
+For example, the user might deny the proof request in their mobile wallet, which would result in a WalletConnect error with a specific error code. We capture this and throw a more descriptive error message to improve the user experience.
+
+Next, let's add helper methods for session management. First, the ``disconnect`` method:
+
+.. code-block:: typescript
+
+  // Disconnect from the wallet
+  async disconnect(): Promise<void> {
+    if (this.topic === undefined) {
+      return;
+    }
+
+    await this.client.disconnect({
+      topic: this.topic,
+      reason: {
+        code: 1,
+        message: "user disconnecting",
+      },
+    });
+  }
+
+The ``disconnect()`` method properly terminates the WalletConnect session:
+
+* It first checks if there's an active session (topic)
+* If no session exists, it returns early without doing anything
+* Otherwise, it sends a disconnect request to the wallet using the WalletConnect client
+* The request includes the session topic and a reason with a standard error code (1) for user-initiated disconnection
+
+After terminating the session with the wallet, we need to update our local state:
+
+.. code-block:: typescript
+
+  // Continuation of disconnect method
+  this.connectedAccount = undefined;
+  this.topic = undefined;
+  super.onAccountChanged(this.connectedAccount);
+
+Once the wallet session is terminated:
+
+* We clear the ``connectedAccount`` property since no account is connected anymore
+* We clear the ``topic`` property since there's no active session
+* We call ``onAccountChanged()`` from our parent class to emit an event notifying the application that the account has changed
+* This ensures that UI components can update to reflect the disconnected state
+
+Now, let's implement the helper for extracting the account address:
+
+.. code-block:: typescript
+
+  // Helper to extract Concordium account from WalletConnect namespaces
+  private getAccount(ns: SessionTypes.Namespaces): string | undefined {
+    const [, , account] =
+      ns[WALLET_CONNECT_SESSION_NAMESPACE].accounts[0].split(":");
+    return account;
+  }
+
+The ``getAccount()`` method handles parsing Concordium account addresses from WalletConnect's namespace format:
+
+* WalletConnect represents accounts in a standardized format: ``namespace:chainId:address``
```

Comment:
Would be good to link to the standard of `walletConnect` where this format is defined.